src/components/checkbox/checkbox.js

```diff
@@ -171,12 +184,18 @@ function MdCheckboxDirective(inputDirective, $mdAria, $mdConstant, $mdTheming, $
       }
 
       function render() {
-        if(ngModelCtrl.$viewValue) {
+        if(ngModelCtrl.$viewValue && !isIndeterminant) {
           element.addClass(CHECKED_CSS);
         } else {
           element.removeClass(CHECKED_CSS);
         }
       }
+
+      function setIndeterminantState() {
+        isIndeterminant = scope.$eval(attr.mdIndeterminant) === false ? false : true;
```

Comment:
Hmm so, the $scope.$watch will pass in the new value as the first parameter which will either be - `undefined`, `true`, or `false`.

If it returns `false` we want to set `isIndeterminate` to `false`. 

All other cases we want to set  `isIndeterminate` to `true`.

I cannot seem to find a better way to achieve that, but I am open to suggestions!